### PR TITLE
Geomap: Refactor marker style options into reusable component

### LINKED ISSUE #41657: Geomap: Separate out marker style UI into reusable component
Separate out marker style UI into a separate, reusable React component (instead of being built with builder options). This is first step of change of unifying style handling across all layer options in Geomap

### PR DESCRIPTION
**Pull request by nmarrs**
**What this PR does / why we need it**:
Refactor out marker style options into a separate reusable component that we can use across all geomap layer types.

PORK:
- Add ability to remove current selected symbol
- Add ability to remove current text
- Improve combined marker / text label layer UX

ezgif com-gif-maker

**Which issue(s) this PR fixes**:

Fixes #41657

### CODE REVIEW

**Comment by ryantxu** [comment]
Looks good -- can we try to add/use the same config on the text layer?

It will require adding the additional controls for text options on the same config object :)

**Comment by nmarrs** [comment]
Updated PR with additional changes switching out text layer style options with new reusable component 👍

**Comment by ryantxu** [comment]
There are two changes that will make this more easy to reuse in various contexts:
image

1. Symbol picker should allow removing the value (ie change button to 'x')
2. The text field... i think we only need the option to set a field, and not the option to enter raw text.

**Comment by nmarrs** [comment]
> 2\. The text field... i think we only need the option to set a field, and not the option to enter raw text.

Not sure on this one, as we are also using the TextDimension component for canvas. Do we want to disable custom text altogether for both geomap / canvas? Also having the ability to have custom text may have some value add in some cases

Addressed having the ability to remove current symbol selection 👍

**Comment by ryantxu** [comment]
how about just make sure the text field also has a way of clearing the selection?

File: public/app/features/dimensions/editors/TextDimensionEditor.tsx [hunk 1]
```diff
@@ -81,12 +88,17 @@ export const TextDimensionEditor: FC<StandardEditorProps<TextDimensionConfig, Te
...
         <InlineFieldRow>
           <InlineField label={'Value'} labelWidth={labelWidth} grow={true}>
-            <StringValueEditor
-              context={context}
-              value={value?.fixed}
-              onChange={onFixedChange}
-              item={dummyStringSettings}
-            />
+            <>
+              <StringValueEditor
```
Comment (ryantxu) [inline public/app/features/dimensions/editors/TextDimensionEditor.tsx:92]: maybe the X can be internal to this component?  maybe add a setting for `isClearable`?

**Review by ryantxu — APPROVED** [review]
looks great!

**Pull request merged**